### PR TITLE
chase: init at 0.5.2

### PR DESCRIPTION
**Pull request by polyrod**
###### Motivation for this change
NixOs handles different versions of software contemporary installed on a NixOs installation.
Dispatching is done via symlinks.
"Chase" automatically resolves  to the last target of a chain of symlinks avoiding looping.
I think its a valuable tool for any NixOs admin and is good company for "which".

###### Things done

Added chase under pkgs/tools/system/chase
Its empowers "which" that is installed in the same pkgs path

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/system/chase/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchurl ,pkgconfig, libatomic_ops , boehmgc , ... }:
```
Comment (joachifm) [inline pkgs/tools/system/chase/default.nix]: Please use a closed parameter list (no `...`)

File: pkgs/tools/system/chase/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+stdenv.mkDerivation rec {
+  name = "chase_0.5.2.orig";
```
Comment (joachifm) [inline pkgs/tools/system/chase/default.nix]: Please move `.orig` to `src.url`

File: pkgs/tools/system/chase/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  buildInputs = [ pkgconfig libatomic_ops boehmgc ] ;
+  src = fetchurl {
+    url = "http://ftp.stw-bonn.de/debian/pool/main/c/chase/${name}.tar.gz";
```
Comment (joachifm) [inline pkgs/tools/system/chase/default.nix]: Please use `mirror://debian`

File: pkgs/tools/system/chase/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  makeFlagsArray="LIBS=-lgc";
+
+  meta = {
+    description = "Follow a symlink and print out its target file";
+    longDescription = ''
+      A commandline program that chases symbolic filesystems links to the original file
+    '';
+    homepage = "https://qa.debian.org/developer.php?login=rotty%40debian.org";
+    license = stdenv.lib.licenses.gpl2Plus;
+    maintainers = [ "Maurizio Di Pietro" ];
```
Comment (joachifm) [inline pkgs/tools/system/chase/default.nix]: Please add a proper entry to `maintainers`.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -4359,6 +4359,10 @@ in
 
   which = callPackage ../tools/system/which { };
 
+  chase = callPackage ../tools/system/chase {
+    inherit boehmgc;
```
Comment (joachifm) [inline pkgs/top-level/all-packages.nix]: `callPackage` is supposed to fill in all missing parameter from the enclosing scope; this explicit `inherit` ought to be unnecessary here.

**Review by joachifm — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/chase/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchurl ,pkgconfig, libatomic_ops , boehmgc }:
+
+stdenv.mkDerivation rec {
+  name = "chase_0.5.2";
```
Comment (LnL7) [inline pkgs/tools/system/chase/default.nix]: the convention is to separate the version and name with a dash

File: pkgs/tools/system/chase/default.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchurl {
+    url = "mirror://debian/pool/main/c/chase/${name}.orig.tar.gz";
+    sha256 = "68d95c2d4dc45553b75790fcea4413b7204a2618dff148116ca9bdb0310d737f";
+  };
+
+  doCheck = true;
+  makeFlags = [ "-e" ];
+  makeFlagsArray="LIBS=-lgc";
+
+  meta = {
```
Comment (LnL7) [inline pkgs/tools/system/chase/default.nix]: you could use `meta = with stdenv.lib;` here so you don't need the full qualified attributes anymore

**Review by LnL7 — APPROVED** [review]
nitpicks

**Review by joachifm — APPROVED** [review]

**Comment by grahamc** [comment]
Merged in e0878ce3163272f097b14f36dd61adca18257799 note I collapsed your commits in to one with the text "chase: init at xxx". I also fixed the hyphen in the name. Thank you!

**Comment by grahamc** [comment]
One other thing -- I had to rename your entry in maintainers.nix because it needs to match your github name. I did that in fa89f84c546b2f7e669cacb4e457e2460a487e0a.

File: pkgs/tools/system/chase/default.nix (hunk 6 above)
Comment (polyrod) [inline pkgs/tools/system/chase/default.nix]: The file on the debian repo seperates name and version with a "_" , so i used it .
Thank you.

**Comment by polyrod** [comment]
Hi graham, 
in e0878ce you changed the hyphen to dash ..... the original file on the debian repo has a hyphen in its name to seperate name and version . Your change broke futchurl downloading src .

**Comment by LnL7** [comment]
Oh, can you create a new pr with a fix?

**Comment by polyrod** [comment]
Ok fixed the hyphen issue and  avoided fully qualified attribute names as you suggested

**Comment by polyrod** [comment]
well Graham Christensen already fixed it in the nixos repo directly ,
should work right now.

2017-01-04 11:31 GMT+01:00 Daiderd Jordan <notifications@github.com>:

**Pull request closed without merging**